UIElements/frontPage.py

```diff
@@ -35,10 +37,11 @@ def setPosReadout(self, xPos, yPos, zPos):
         self.yReadoutPos = str(yPos) + " mm"
         self.zReadoutPos = str(zPos) + " mm"
     
-    def setupQueues(self, message_queue, gcode_queue, quick_queue):
-        self.message_queue = message_queue
-        self.gcode_queue = gcode_queue
-        self.quick_queue = quick_queue
+    def setUpData(self, data):
+        self.data = data
+        print "Initialized: " + str(self)
+        self.gcodecanvas.setUpData(data)
```

Comment:
Should these be removed from individual instances and use a "for child in children" type argument or call an "self.initialize" function from within the makesmith init function?